sascha/iban/script.js

```diff
@@ -0,0 +1,176 @@
+            numbers.push(arrToNumbers[i].charCodeAt(0)-55);  
+        }
+        numbers = numbers.join('');
+        return numbers;
+       
+    }
+    
+    //This function grabs the account number and checks whether it's length is equal to 10, it will add zeroes in front if it does not have enough
+    var bankRekeningLengthChecker = function(acc){
+               
+        //checks whether the length of the string is equal to 10, if it is it returns the number
+        if(acc.length == 10){
+            return acc;
+        }
+        //if it's smaller than 10 it adds zeroes at the beginning so that the total input is converted to 10 numbers
+        else if(acc.length < 10){
+            //it calculates the difference between 10 and the length of the input and unshifts 0's in at the beginning based on that difference
+            var temp = 10-acc.length;
+            acc = acc.split("");
+				
+            for(var i = 0; i < temp;i++){
+                acc.unshift('0');
+            }
+            acc = acc.join("");
+            return acc;
+            }
+        else if(acc.length > 10){
+            return warningMessage("Ongeldige invoer! Vul a.u.b. een bankrekening onder de 10 cijfers in!");
+        }    
+    }
+   
+    //--------------Getters------------------------------------------------------------------------------------
+    //this function grabs the landcode and injects it into the numbers function
+    var getLandCode = function(){
+        return lettersToNumbers(landCode);
+    }
+    //this function grabs the bankCode which was selected by the user and magically turns it into numbers
+    var getBankCode = function(){
+        return lettersToNumbers(bankCode);
+    }
+    //TODO bankrekeningchecker herschrijven zodat hij eerst dingen check maar wel altijd een value teruggeeft
+    //this function grabs 
+    var getRektnummer = function(){
+        return bankRekeningLengthChecker(bankRekening);    
+    }
+    
+    //gets the whole iban input no, so that the conrol no can be calculated
+    var getWholeIban = function(lcode, bcode, brek){
+        var list = "";
+        
+        list += bcode;
+        list += brek;
+        list += lcode;
+		list += '00';
+        return list;
+    }
+    
+    
+        //deze functie past mod97 toe op bankcode + bankrekening + landcode + 00 en trekt het af van 98 voor het controlegetal
+    function mod97(iban) {
+    var remainder = iban;
+    var checkNo;
+    var block;
+
+        while (remainder.length > 2) {
+            block = remainder.slice(0, 9);
+            remainder = parseInt(block, 10) % 97 + remainder.slice(block.length);
+        }
+        
+        return checkNo = (parseInt(remainder, 10) % 97);
+	}
+	function generateControleGetal(iban){
+        iban = 98- iban; 
+		if(iban < 10){
+            iban = iban.toString();
+            iban = iban.split("");
+            iban.unshift(0);
+            iban = iban.join("");
+        }
+        return iban;

```

Comment:
let op: als iban >= 10, dan is het een getal dat je verder niet aanpast en retourneert.
als iban <=10, dan maak je er een string van, zet er een 0 voor, en retourneert die.
deze functie kan dus zowel een getal als een string retourneren, en je weet niet wat het wordt als je hem aanroept.

Dat is een valkuil van JavaScript, in deze opgave gaat het goed, maar het kan ook makkelijk verkeerd gaan, en is dan heel lastig de debuggen. Veiliger is dus altijd hetzelfde type te retourneren, in dit geval een string met `.toString();`